packages/plugin-react/src/jsx-runtime/babel-restore-jsx.ts

```diff
@@ -36,7 +46,7 @@ export default function ({ types: t }: typeof babel): babel.PluginObj {
     }
 
     const props = getJSXProps(propsNode)
-    if (props == null) {
+    if (props == null || isInvalidProps(props)) {
```

Comment:
Why shouldn't we just remove any `__self` and `__source` props found, instead of bailing out?

Reply:
> Why shouldn't we just remove any `__self` and `__source` props found, instead of bailing out?

Before I thought we just needed to avoid broken, but now I think it's ok to remove `__self` and `__source`